Splunk addon name change and related updates (#62673)
openshift/openshift-docs: modules/install-and-configure-the-splunk-add-on.adoc

//
// * integration/integrate-with-splunk.adoc
:_content-type: PROCEDURE
[id="install-and-configure-the-splunk-add-on_{context}"]
= Installing and configuring the Splunk add-on

You can install the {product-title} add-on from your Splunk instance.

[NOTE]
====
To maintain backward compatibility, the `source_type` and `input_type` parameters for configured inputs are still called `stackrox_compliance`, `stackrox_violations`, and `stackrox_vulnerability_management`.
====

.Prerequisites
* You must have an API token with `read` permission for all resources of {product-title}. You can assign the *Analyst* system role to grant this level of access. The *Analyst* role has read permissions for all resources.

.Procedure
. Download the {product-title} add-on from link:https://splunkbase.splunk.com/app/5315/[Splunkbase].
. Navigate to the Splunk home page on your Splunk instance.
. Navigate to *Apps* -> *Manage Apps*.
. Select *Install app from file*.
. In the *Upload app* pop-up box, select *Choose File* and select the {product-title} add-on file.
. Click *Upload*.
. Click *Restart Splunk*, and confirm to restart.
. After Splunk restarts, select *{product-title}* from the *Apps* menu.

> . Go to *Configuration* and then click *Add-on Settings*.
> . For *Central Endpoint*, enter the IP address or the name of your Central instance. For example, `central.custom:443`.
> . Enter the *API token* you have generated for the add-on.
> . Click *Save*.

Reviewer: ```suggestion
. Navigate to *Configuration > Add-on Settings*:
.. For *Central Endpoint*, enter the IP address or the name of your Central instance. For example, enter `central.custom:443`.
.. Enter the *API token* you have generated for the add-on.
.. Click *Save*.
```

Reviewer 2: Good suggestion of "Navigate", but it is `->` for the arrow

Author: Thank you @agantony However in the UI, users have to click *Configuration* then on the page that opens they have to click on *Add-on Settings*. Since, this is not a direct menu option I kept it separate.

. Go to *Inputs*.

> . Click *Create New Input*.

Reviewer: ```suggestion
. Click *Create New Input*, and select one of the following:
```

> . Select *ACS Compliance* to pull compliance data, *ACS Violations* to pull violation data, or *ACS Vulnerability Management* to pull vulnerability data into Splunk.

Reviewer: ```suggestion
.. *ACS Compliance* 
+
Pulls compliance data.
.. *ACS Violations*
+
Pulls violation data.
.. *ACS Vulnerability Management*
+
Pulls vulnerability data into Splunk.
```

Reviewer 2: I like this suggestion to put the options into bullets, but they should be `**` to be unnumbered

. Enter a *Name* for the input.
. Select an *Interval* to pull data from {product-title}.
For example, every 14400 seconds.
. Select the Splunk *Index* to which you want to send the data.
. For *Central Endpoint*, enter the IP address or the name of your Central instance.
. Enter the *API token* you have generated for the add-on.
. Click *Add*.

.Verification

> To verify the the {product-title} add-on installation, query the received data.

Reviewer 2: Not sure if we actually say not to, but the verification section is usually steps. Since we don't do lead-in sentences for prereqs and the procedure, probably should also stick to just steps for the verification section as well.

What I usually do is something like:

```
* To verify the the {product-title} add-on installation, query the received data.
.. In you Splunk instance, go to to *Search* and type `index=* sourcetype="stackrox-*"` as the query.
.. Press *Enter*.
```

> . In you Splunk instance, go to to *Search* and type `index=* sourcetype="stackrox-*"` as the query.

Reviewer: ```suggestion
. In your Splunk instance, navigate to *Search* and type `index=* sourcetype="stackrox-*"` as the query.
```

. Press *Enter*.

> All your configured sources must show up in search results.

Reviewer: ```suggestion
All your configured sources must be displayed in the search results.
```

> All your configured sources must show up in search results.

Reviewer 2: I'm not sure if I understand what this sentence means, that the sources MUST show up. Should it instead be something like "Verify that your configured sources are displayed in the search results."
